docs/docs/spark-procedures.md

```diff
@@ -972,4 +972,91 @@ CALL catalog_name.system.compute_table_stats(table => 'my_table', snapshot_id =>
+The newly rewritten metadata files enable moving or coping an Iceberg table to a new location.
+After copying both metadata and data to the desired location, the replicated iceberg
+table will appear identical to the source table, including snapshot history, schema and partition specs.
+
+!!! info
+    This procedure only creates metadata for an existing Iceberg table modified for a new location. The produced file_list can be used for copying rewritten metadata and data files to the new location.
+    Copying/Moving metadata and data files to the new location is not part of this procedure.
+
+
+| Argument Name      | Required? | default                                        | Type   | Description                                                             |
+|--------------------|-----------|------------------------------------------------|--------|-------------------------------------------------------------------------|
+| `table`            | ✔️        |                                                | string | Name of the table                                                       |
+| `source_prefix`    | ✔️        |                                                | string | The existing prefix to be replaced                                      |
+| `target_prefix`    | ✔️        |                                                | string | The replacement prefix for `source_prefix`                              |
+| `start_version`    |           | first metadata.json in table's metadata log    | string | The name or path to the chronologically first metadata.json to rewrite. |
+| `end_version`      |           | latest metadata.json                           | string | The name or path to the chronologically last metadata.json to rewrite   |
+| `staging_location` |           | new directory under table's metadata directory | string | The output location for newly modified metadata files                   |
+
+
+#### Modes of operation:
+
+- Full Rewrite:
+
+By default, the procedure operates in full rewrite mode, rewriting all reachable metadata files. This includes metadata.json, manifest lists, manifests, and position delete files.
+
+- Incremental Rewrite:
+
+If `start_version` is provided, the procedure will only rewrite metadata files created between `start_version` and `end_version`. `end_version` defaults to the latest metadata location of the table. 
+
+#### Output
+
+| Output Name          | Type   | Description                                                                         |
+|----------------------|--------|-------------------------------------------------------------------------------------|
+| `latest_version`     | string | Name of the latest metadata file rewritten by this procedure                        |
+| `file_list_location` | string | Path to a file containing a listing of comma-separated source and destination paths |
```

Comment:
I feel we did some rounds in the javadoc for the Result, so would take from that?

```
Path to a file containing a copy plan for all files added to the table between startVersion and endVersion, comma-separated.

For each file, it will include the source path (either the original path in the table, or in the staging location if rewritten), and the target path (under the new prefix).
 
This allows for copying a functioning version of the table to the target prefix.
```

Reply:
Sounds great! I think we can keep short description in the table to be concise, I added a section below for file list which highlight the source and target prefix and examples below with more details as you suggested

Reply:
How about "Path to a csv file containing a mapping of source and target paths"?
* csv file is comma-separated by nature
* `target paths` to be consistent with `target prefix"